File: h.py
import calendar
import datetime

now = datetime.datetime.now()
dotw = calendar.day_name[now.weekday()]

y = int(now.strftime("%Y"))
m = int(now.strftime("%m"))
d = int(now.strftime("%d"))

def swimLessons():
  sl1 = datetime.datetime(year=2023, month=6, day=5, hour=00, minute=00) <= now < datetime.datetime(year=2023, month=6, day=9, hour=00, minute=00)

  sl2 = datetime.datetime(year=2023, month=6, day=12, hour=00, minute=00) <= now < datetime.datetime(year=2023, month=6, day=16, hour=00, minute=00)
  
  sl3 = datetime.datetime(year=2023, month=6, day=19, hour=00, minute=00) <= now < datetime.datetime(year=2023, month=6, day=23, hour=00, minute=00)
  
  sl4 = datetime.datetime(year=2023, month=6, day=26, hour=00, minute=00) <= now < datetime.datetime(year=2023, month=6, day=30, hour=00, minute=00)
  
  sl5 = datetime.datetime(year=2023, month=7, day=10, hour=00, minute=00) <= now < datetime.datetime(year=2023, month=7, day=14, hour=00, minute=00)
  
  sl6 = datetime.datetime(year=2023, month=7, day=17, hour=00, minute=00) <= now < datetime.datetime(year=2023, month=7, day=21, hour=00, minute=00)
  
  sl7 = datetime.datetime(year=2023, month=7, day=24, hour=00, minute=00) <= now < datetime.datetime(year=2023, month=7, day=28, hour=00, minute=00)
  
  sl8 = datetime.datetime(year=2023, month=7, day=31, hour=00, minute=00) <= now < datetime.datetime(year=2023, month=8, day=4, hour=00, minute=00)

  if (sl1 == True) or (sl2 == True) or (sl3 == True) or (sl4 == True) or (sl5 == True) or (sl6 == True) or (sl7 == True) or (sl8 == True):
    return True

def amTimes(am_open, am_close):
  print(f"OPEN:")
  print(f"Open Swim (ALL AGES)")
  print(am_open.strftime("%I") + ":" + am_open.strftime("%M"), am_open.strftime("%p"), "-", am_close.strftime("%I") + ":" + am_close.strftime("%M"), am_close.strftime("%p"))

def familyNight(family_night_open, family_night_close):
  print("OPEN:")
  print("Family Night - $15 per family (ALL AGES)")
  print(family_night_open.strftime("%I") + ":" + family_night_open.strftime("%M"), family_night_open.strftime("%p"), "-", family_night_close.strftime("%I") + ":" + family_night_close.strftime("%M"), family_night_close.strftime("%p"))

def halfOff(half_off_open, half_off_close):
  print("OPEN:")
  print("Open Swim - Half Price! (ALL AGES)")
  print(half_off_open.strftime("%I") + ":" + half_off_open.strftime("%M"), half_off_open.strftime("%p"), "-", half_off_close.strftime("%I") + ":" + half_off_close.strftime("%M"), half_off_close.strftime("%p"))

if datetime.datetime(year=2023, month=6, day=2, hour=00, minute=00) <= now < datetime.datetime(year=2023, month=9, day=1, hour=00, minute=00):
#SUNDAY
   if dotw == "Sunday":
     am_open = datetime.datetime(year=y, month=m, day=d, hour=12, minute=00)
     am_close = datetime.datetime(year=y, month=m, day=d, hour=17, minute=45)
     family_night_open = datetime.datetime(year=y, month=m, day=d, hour=18, minute=30)
     family_night_close = datetime.datetime(year=y, month=m, day=d, hour=20, minute=30)
     if am_open <= now < am_close:
       amTimes(am_open, am_close)
     elif am_close <= now < family_night_open:
       print("CLOSED:")
       print("Family Night starts at 6:30 PM")
     elif family_night_open <= now < family_night_close:
       familyNight(family_night_open, family_night_close)
     elif family_night_close <= now:
       print("CLOSED")
#SATURDAY
   elif dotw == "Saturday":
     am_open = datetime.datetime(year=y, month=m, day=d, hour=11, minute=30)
     am_close = datetime.datetime(year=y, month=m, day=d, hour=18, minute=00)
     if am_open <= now < am_close:
       amTimes(am_open, am_close)
     elif am_close <= now:
       print("CLOSED")
#M, T, TH
   elif (dotw == "Monday") or (dotw == "Tuesday") or (dotw == "Thurday"):
     am_open = datetime.datetime(year=y, month=m, day=d, hour=11, minute=30)
     if swimLessons() == True:
       am_close = datetime.datetime(year=y, month=m, day=d, hour=17, minute=30)
       if am_open <= now < am_close:
         amTimes(am_open, am_close)
       elif am_close <= now:
         print("CLOSED")
     elif swimLessons() != True:
       am_close = datetime.datetime(year=y, month=m, day=d, hour=19, minute=00)
       half_off_open = datetime.datetime(year=y, month=m, day=d, hour=17, minute=00)
       half_off_close = datetime.datetime(year=y, month=m, day=d, hour=19, minute=00)
       if am_open <= now < am_close:
         amTimes(am_open, am_close)
       elif half_off_open <= now < half_off_close:
         halfOff(half_off_open, half_off_close)
       elif half_off_close <= now:
         print("CLOSED")
#FRIDAY
   elif dotw == "Friday":
     am_open = datetime.datetime(year=y, month=m, day=d, hour=11, minute=30)
     am_close = datetime.datetime(year=y, month=m, day=d, hour=19, minute=00)
     half_off_open = datetime.datetime(year=y, month=m, day=d, hour=17, minute=00)
     half_off_close = datetime.datetime(year=y, month=m, day=d, hour=19, minute=00)
     if am_open <= now < am_close:
       amTimes(am_open, am_close)
     elif half_off_open <= now < half_off_close:
       halfOff(half_off_open, half_off_close)
     elif half_off_close <= now:
       print("CLOSED")
#WEDNESDAY
   elif dotw == "Wednesday":
     am_open = datetime.datetime(year=y, month=m, day=d, hour=12, minute=00)
     if swimLessons() == True:
       am_close = datetime.datetime(year=y, month=m, day=d, hour=17, minute=30)
       if am_open <= now < am_close:
         amTimes(am_open, am_close)
       elif am_close <= now:
         print("CLOSED")
     elif swimLessons() != True:
       am_close = datetime.datetime(year=y, month=m, day=d, hour=17, minute=45)
       family_night_open = datetime.datetime(year=y, month=m, day=d, hour=18, minute=30)
       family_night_close = datetime.datetime(year=y, month=m, day=d, hour=20, minute=30)
       if am_open <= now < am_close:
         amTimes(am_open, am_close)
       elif am_close <= now < family_night_open:
         print("CLOSED:")
         print("Family Night starts at 6:30 PM")
       elif family_night_open <= now < family_night_close:
         familyNight(family_night_open, family_night_close)
       elif family_night_close <= now:
         print("CLOSED")

File: h.js
var now = new Date();
var dotw = now.toLocaleString('en-us', { weekday: 'long' });
var y = now.getFullYear();
var m = now.getMonth();
var d = now.getDate();

function swimLessons() {
  const sl1 = new Date(2023, 5, 5) <= now && now < new Date(2023, 5, 9);
  const sl2 = new Date(2023, 5, 12) <= now && now < new Date(2023, 5, 16);
  const sl3 = new Date(2023, 5, 19) <= now && now < new Date(2023, 5, 23);
  const sl4 = new Date(2023, 5, 26) <= now && now < new Date(2023, 5, 30);
  const sl5 = new Date(2023, 6, 10) <= now && now < new Date(2023, 6, 14);
  const sl6 = new Date(2023, 6, 17) <= now && now < new Date(2023, 6, 21);
  const sl7 = new Date(2023, 6, 24) <= now && now < new Date(2023, 6, 28);
  const sl8 = new Date(2023, 6, 31) <= now && now < new Date(2023, 7, 4);
  if (sl1 || sl2 || sl3 || sl4 || sl5 || sl6 || sl7 || sl8) {
    return true;
  }
}

function amTimes(am_open, am_close) {
    document.getElementById("h-head").innerHTML = "THE POOL IS CURRENTLY <u>OPEN</u>:";
    document.getElementById("h-body").innerHTML = "<u>Open Swim (ALL AGES)</u><br><b>" + am_open.toLocaleString('en-US', { hour: 'numeric', minute: 'numeric', hour12: true }) + " - " + am_close.toLocaleString('en-US', { hour: 'numeric', minute: 'numeric', hour12: true }) + "</b>";
}

function familyNight(family_night_open, family_night_close) {
    document.getElementById("h-head").innerHTML = "THE POOL IS CURRENTLY <u>OPEN</u>:";
    document.getElementById("h-body").innerHTML = "<u>Family Night - $15 per family (ALL AGES)</u><br><b>"+ family_night_open.toLocaleString('en-US', { hour: 'numeric', minute: 'numeric', hour12: true }) + " - " + family_night_close.toLocaleString('en-US', { hour: 'numeric', minute: 'numeric', hour12: true }) + "</b>";
}

function halfOff(half_off_open, half_off_close) {
    document.getElementById("h-head").innerHTML = "THE POOL IS CURRENTLY <u>OPEN</u>:";
    document.getElementById("h-body").innerHTML = "<u>Open Swim - Half Price! (ALL AGES)</u><br><b>" + half_off_open.toLocaleString('en-US', { hour: 'numeric', minute: 'numeric', hour12: true }) + " - " + half_off_close.toLocaleString('en-US', { hour: 'numeric', minute: 'numeric', hour12: true }) + "</b>";
}

function morning(am_open) {
  document.getElementById("h-head").innerHTML = "THE POOL IS CURRENTLY <u>CLOSED</u>";
  document.getElementById("h-body").innerHTML = "We will be open at " + am_open.toLocaleString('en-US', {
    hour: 'numeric',
    minute: 'numeric',
    hour12: true
  });
}

if (new Date(2023, 5, 2) <= now && now < new Date(2023, 8, 1)) {
  if (dotw === "Sunday") {
    var am_open = new Date(y, m, d, 12, 0);
    var am_close = new Date(y, m, d, 17, 45);
    var family_night_open = new Date(y, m, d, 18, 30);
    var family_night_close = new Date(y, m, d, 20, 30);
    if (am_open <= now && now < am_close) {
      amTimes(am_open, am_close);
    } else if (am_close <= now && now < family_night_open) {
        document.getElementById("h-head").innerHTML = "THE POOL IS CURRENTLY <u>CLOSED</u>";
        document.getElementById("h-body").innerHTML = "Family Night starts at 6:30 PM";
    } else if (family_night_open <= now && now < family_night_close) {
      familyNight(family_night_open, family_night_close);
    } else if (family_night_close <= now) {
        document.getElementById("h-head").innerHTML = "THE POOL IS CURRENTLY <u>CLOSED</u>";
    } else if (now < am_open) {
      morning(am_open);
    }
  } else if (dotw === "Saturday") {
    var am_open = new Date(y, m, d, 11, 30);
    var am_close = new Date(y, m, d, 18, 0);
    if (am_open <= now && now < am_close) {
      amTimes(am_open, am_close);
    } else if (am_close <= now) {
        document.getElementById("h-head").innerHTML = "THE POOL IS CURRENTLY <u>CLOSED</u>";
    } else if (now < am_open) {
      morning(am_open);
    }
  } else if (dotw === "Monday" || dotw === "Tuesday" || dotw === "Thursday") {
    var am_open = new Date(y, m, d, 11, 30);
    if (swimLessons()) {
      var am_close = new Date(y, m, d, 17, 30);
      if (am_open <= now && now < am_close) {
        amTimes(am_open, am_close);
      } else if (am_close <= now) {
        document.getElementById("h-head").innerHTML = "THE POOL IS CURRENTLY <u>CLOSED</u>";
      } else if (now < am_open) {
        morning(am_open);
      }
    } else {
      var am_close = new Date(y, m - 1, d, 19, 0);
      var half_off_open = new Date(y, m - 1, d, 17, 0);
      var half_off_close = new Date(y, m - 1, d, 19, 0);
      if (am_open <= now && now < am_close) {
        amTimes(am_open, am_close);
      } else if (half_off_open <= now && now < half_off_close) {
        halfOff(half_off_open, half_off_close);
      } else if (half_off_close <= now) {
        document.getElementById("h-head").innerHTML = "THE POOL IS CURRENTLY <u>CLOSED</u>";
      } else if (now < am_open) {
        morning(am_open);
      }
    }
  } else if (dotw === "Friday") {
    var am_open = new Date(y, m, d, 11, 30);
    var am_close = new Date(y, m, d, 19, 0);
    var half_off_open = new Date(y, m, d, 17, 0);
    var half_off_close = new Date(y, m, d, 19, 0);
    if (am_open <= now && now < am_close) {
      amTimes(am_open, am_close);
    } else if (half_off_open <= now && now < half_off_close) {
      halfOff(half_off_open, half_off_close);
    } else if (half_off_close <= now) {
        document.getElementById("h-head").innerHTML = "THE POOL IS CURRENTLY <u>CLOSED</u>";
    } else if (now < am_open) {
      morning(am_open);
    }
  } else if (dotw === "Wednesday") {
    var am_open = new Date(y, m, d, 11, 30);
    if (swimLessons()) {
      var am_close = new Date(y, m, d, 17, 30);
      if (am_open <= now && now < am_close) {
        amTimes(am_open, am_close);
      } else if (am_close <= now) {
        document.getElementById("h-head").innerHTML = "THE POOL IS CURRENTLY <u>CLOSED</u>";
      } else if (now < am_open) {
        morning(am_open);
      }
    } else {
      var am_close = new Date(y, m, d, 17, 45);
      var family_night_open = new Date(y, m, d, 18, 30);
      var family_night_close = new Date(y, m, d, 20, 30);
      if (am_open <= now && now < am_close) {
        amTimes(am_open, am_close);
      } else if (am_close <= now && now < family_night_open) {
        document.getElementById("h-head").innerHTML = "THE POOL IS CURRENTLY <u>CLOSED</u>";
        document.getElementById("h-body").innerHTML = "Family Night starts at 6:30 PM";
      } else if (family_night_open <= now && now < family_night_close) {
        familyNight(family_night_open, family_night_close);
      } else if (family_night_close <= now) {
        document.getElementById("h-head").innerHTML = "THE POOL IS CURRENTLY <u>CLOSED</u>";
      } else if (now < am_open) {
        morning(am_open);
      }
    }
  }
}